File: wngbuild/wngbuild.rb
class BuildProfile
  def initialize(files, output)
    @files = files
    @output = output
    @cc = "gcc"
    @flags = ""
  * Parameters
  def cc=(cc)
    @cc=cc    
  end
  def cc
    @cc
  end
  def flags=(flags)
    @flags=flags
  end
  * Methods  
  def run
    system(@cc, @files, "-o", @output, @flags)
  end
  def runOutput
    system("./#{@output}")
  end
end

File: wngbuild/wngbuild.py
import subprocess as sp

class BuildProfile:
    def __init__(self, files, output):
        self.files = files
        self.output = output
        self.cc = "gcc"
        self.flags = ""

    def run(self):
        cmd = f"{self.cc} {self.files} -o {self.output} {self.flags}"
        sp.call(cmd)

    def runOutput(self):
        cmd = f"./{self.output}"
        sp.call(cmd)
